mantidproject/mantid
# Add outline RegionSelector widget and refactor SliceViewer

This PR refactors SliceViewer to move the `dataview` to its own file and separate it from the rest of SliceViewer so that it can more easily be embedded on separate widgets. This PR also adds an outline `RegionSelector` widget which embeds the new SliceViewer component. This essentially contains SliceViewer functionality excluding the peaks viewer functionality and will be expanded in future work as part of #33037 to add multiple region selection components.

Quite a lot of other refactoring has taken place to remove reliance on the `view` class and improve testability, and some tidying has been done to move files into subdirectories.

**To test:**

- A test interface has been added to demonstrate embedding the `dataview` into the outline `RegionSelector` widget but this needs to be enabled in your user properties file. To do this, add the following to your `Mantid.user.properties` file (or just add `Utility/RegionSelectorInterface.py` to this line if it already exists). Note you'll need to restart Mantid for this to take effect. Make sure you remove it afterwards because this will lose the other python interfaces that you get by default unless you also list them.
```
mantidqt.python_interfaces=Utility/RegionSelectorInterface.py
```
- The test interface is very basic and requires a workspace to be loaded first, so load any workspace that can be used with SliceViewer, e.g. `ws=Load('INTER45455')`.
- Go to the Interfaces menu and under Utility click on RegionSelectorInterface. This will show a plot that looks very much like SliceViewer but the peaks button will do nothing.
- **More importantly**, since this changes quite a lot under-the-hood on SliceViewer, test SliceViewer thoroughly to check it is still functioning as expected. The [manual testing instructions](https://developer.mantidproject.org/Testing/SliceViewer/SliceViewer.html) may be helpful

Fixes #33038 

*This does not require release notes* because **it is under-the-hood work that is only visible in a non-user-facing test interface**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
